src/projection.ts

```diff
@@ -137,6 +134,21 @@ export class AtlasProjection extends BaseAtlasClass<
     }
   }
 
+  async datasetId(): Promise<UUID> {
+    if (this.project_id !== undefined) {
+      return this.project_id;
+    }
+    if (this._project !== undefined) {
+      return this._project.id;
+    }
+    const endpoint = `/v1/project/projection/${this.id}/get/dataset_id`;
+    const { dataset_id } = (await this.apiCall(endpoint, 'GET')) as {
+      dataset_id: UUID;
+    };
+    this.project_id = dataset_id;
+    return dataset_id;
```

Comment:
This should be wrapped into a `Promise<string> | null` at `this._project_id` or something similar; otherwise we might trigger multiple concurrent fetches to this same endpoint before the first resolves.

Reply:
Addressed after help from @RLesser (thanks!). Doing undefined instead of null to match rest of library + avoid adding null checks everywhere